service/start.go

```diff
@@ -119,6 +115,132 @@ func (svc *service) startNostr(ctx context.Context, encryptionKey string) error
 	return nil
 }
 
+type createAppSubscriber struct {
+	events.EventSubscriber
+	svc   *service
+	relay *nostr.Relay
+}
+
+func (s *createAppSubscriber) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	switch event.Event {
+	case "app_created":
+		properties, ok := event.Properties.(map[string]interface{})
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+			return
+		}
+		walletChildPubkey, ok := properties["walletChildPubkey"].(string)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app walletChildPubkey")
+			return
+		}
+		id, ok := properties["id"].(uint)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to get app id")
+			return
+		}
+		if walletChildPubkey != "" {
+			go func() {
+				appWalletPrivKey, err := s.svc.keys.GetBIP32ChildKey(uint32(id))
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to calculate app wallet priv key")
+				}
+				err = s.svc.startAppWalletSubscription(ctx, s.relay, walletChildPubkey, appWalletPrivKey)
+				if err != nil {
+					logger.Logger.WithError(err).Error("Failed to subscribe to wallet")
+				}
+				// TODO what should we do?
```

Comment:
I think this is fine to just log an error - if the relay is not connectable, the next time the relay connects the relay subscription will be setup again